src/main/java/bio/terra/service/resourcemanagement/ResourceService.java

```diff
@@ -73,13 +81,30 @@ public GoogleBucketResource getOrCreateBucketForFile(Dataset dataset,
             billingProfile,
             null);
 
-        return bucketService.getOrCreateBucket(
-            dataLocationSelector.bucketForFile(datasetName, billingProfile),
-            projectResource,
+        List<UUID> bucketsForDataset = datasetBucketDao.getBucketForDatasetId(UUID.fromString(datasetId));
+        String bucketName = dataLocationSelector.bucketForFile(datasetId, billingProfile);
+
+        List<GoogleBucketResource> bucketsForBillingProfile = bucketsForDataset.stream()
+            .map(bucketUUID -> lookupBucket(bucketUUID.toString()))
```

Comment:
`lookupBucket()` is another case where we have a `UUID`, which the method wants, but we have to convert it to a `String` and then `lookupBucket()` must convert it back. If it's not easy to make all callers pass a `UUID`, you could add a method overload to `lookupBucket()` to avoid the extra conversion in this case at least.

Reply:
it looks like in most of the places where `lookupBucket()` is being used, the uuid is already a string (ie. in the `GcsPdao`). i did add a method overload to address this though.